lightningd/invoice.c

```diff
@@ -148,6 +149,19 @@ static void json_invoice(struct command *cmd,
 			     INVOICE_MAX_LABEL_LEN);
 		return;
 	}
+	/* description */
+	desc_val = tal_strndup(cmd, buffer + desc->start,
+			       desc->end - desc->start);
+	if (desc->end - desc->start >= BOLT11_FIELD_BYTE_LIMIT) {
```

Comment:
Since we now have parsed `desc_val` we could just use `strlen` here, or invert the allocation with the check so that we safe the allocation if the string is too long anyway.